packages/sdk/src/mb-impl/token.ts

```diff
@@ -0,0 +1,214 @@
+    const tokenId = transfer[0].tokenId;
+
+    return {
+      contractAddress: nftContractId,
+      methodName: METHOD_NAMES.TRANSFER,
+      args: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        receiver_id: receiverId,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        token_id: tokenId,
+      },
+    };
+  }
+};
+
+export const burn = (args: BurnArgs): TransactionArgs => {
+  const { nftContractId, tokenIds } = args;
+
+  return {
+    contractAddress: nftContractId,
+    methodName: METHOD_NAMES.BATCH_BURN,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      token_ids: tokenIds,
+    },
+  };
+};
+
+export const deployContract = (
+  args: DeployTokenContractArgs,
+): TransactionArgs => {
+  const {
+    name,
+    factoryContractId,
+    network = Network.TESTNET,
+    ownerId,
+    metadata,
+  } = args;
+
+  const { symbol, icon, baseUri, reference, referenceHash } = metadata;
+
+  const data = {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    owner_id: ownerId,
+    metadata: {
+      spec: 'nft-1.0.0',
+      name: name.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
+      symbol: symbol.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
+      icon: icon ?? DEFAULT_MB_LOGO,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      base_uri: baseUri ?? null,
+      reference: reference ?? null,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      reference_hash: referenceHash ?? null,
+    },
+  };
+
+  return {
+    contractAddress: factoryContractId || getFactoryContract(network),
+    methodName: METHOD_NAMES.DEPLOY_TOKEN_CONTRACT,
+    args: data,
+  };
+};
+
+export const transferContractOwnership = (
+  args: TransferTokenContractOwnership,
+): TransactionArgs => {
+  const { nftContractId, nextOwner, options } = args;
+  const { keepMinters = true } = options;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      new_owner: nextOwner,
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      keep_old_minters: keepMinters,
+    },
+    methodName: METHOD_NAMES.TRANSFER_TOKEN_CONTRACT_OWNERSHIP,
+  };
+};
+
+export const mint = (args: MintArgs): TransactionArgs => {
+  const { nftContractId, options } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {},
+    methodName: METHOD_NAMES.MINT,
+  };
+};
+
+// TODO: do we want this method? How can we reuse `mint` instead of having an extra method
+export const mintMore = (): void => {
+  return;
+};
+
+export const addMinter = (args: AddRemoveMinterArgs): TransactionArgs => {
+  const { minterId, nftContractId } = args;
+
+  return {
+    contractAddress: nftContractId,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      account_id: minterId,
+    },
+    methodName: METHOD_NAMES.ADD_MINTER,
```

Comment:
Shouldn't we just use `batch_change_minters` for all cases so we can deprecate the old methods?

Reply:
Yes, if that is your plan on the contract level, let me know.